src/sentry/incidents/logic.py

```diff
@@ -910,7 +910,7 @@ def delete_alert_rule(alert_rule, user=None):
     with transaction.atomic():
         incidents = Incident.objects.filter(alert_rule=alert_rule)
         bulk_delete_snuba_subscriptions(list(alert_rule.snuba_query.subscriptions.all()))
-        if incidents:
+        if incidents.exists():
```

Comment:
currently lazy `incidents` queryset is evaluated, resulting in every incident being read from the db. This simply checks if at least one row exists in db.